torch/csrc/jit/passes/alias_analysis.h

```diff
@@ -78,21 +78,37 @@ class AliasDb {
       return false;
     }
 
-    T<Element*> aElements;
-    for (const Value* v : a) {
-      if (elementMap_.count(v)) {
-        aElements.insert(elementMap_.at(v));
+    // Record all memory locations from group `a`
+    MemoryLocations memoryLocations;
+    for (auto it = a.cbegin(); it != a.cend();) {
+      const auto value = *it;
+      if (elementMap_.count(value)) {
+        auto element = elementMap_.at(value);
+        memoryLocations |= element->getMemoryLocations();
       }
+
+      do {
+        ++it;
+      } while (it != a.cend() && *it == value);
```

Comment:
Is it true that for an arbitrary unordered multiset, elements with the same key will be located contiguously when you iterate?

Reply:
@suo Thoughts here?  Doing it this way was measurably faster, and I'm not clear on what the actual multiset requirements are here.

Reply:
That said, I don't *think* this differs from the original implementation.  Using std::advance also assumed they were contiguous?

Reply:
yes

Reply:
to both questions haha

Reply:
Ah that's true that it's the same API as before. LGTM then.